edb/repl/__init__.py

```diff
@@ -289,17 +290,85 @@ def command_connect(self, args: str) -> None:
 
     @_command('l', R'\l', 'list databases')
     def command_list_dbs(self, args: str) -> None:
-        result, _ = self.fetch(
-            '''
-                SELECT name := sys::Database.name
-                ORDER BY name ASC
-            ''',
-            json=False
+        self._command_simplelist(
+            '',
+            query='SELECT name := sys::Database.name',
+            name_field='name',
+            item_name='databases',
         )
 
-        print('List of databases:')
-        for dbn in result:
-            print(f'  {dbn}')
+    @_command('lr', R'\lr [PATTERN]', 'list roles')
+    def command_list_roles(self, args: str) -> None:
+        self._command_simplelist(
+            args.strip(),
+            query='SELECT name := sys::Role.name',
+            name_field='name',
+            item_name='roles',
+        )
+
+    @_command('lrI', R'\lrI [PATTERN]',
+              'list roles (case sensitive pattern match)')
+    def command_list_roles_i(self, args: str) -> None:
+        self._command_simplelist(
+            args.strip(),
+            query='SELECT name := sys::Role.name',
+            name_field='name',
+            item_name='roles',
+            case_sensitive=True,
```

Comment:
To avoid code dup please follow what we did for the `command_describe_object` method

Reply:
OK, I've used the same pattern for "I" flag versions of the specific `\l` commands.